atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/QosCassandraClient.java

```diff
@@ -69,15 +69,14 @@ public QosCassandraClient(CassandraClient client, QosClient qosClient) {
     public Map<ByteBuffer, List<ColumnOrSuperColumn>> multiget_slice(String kvsMethodName, TableReference tableRef,
             List<ByteBuffer> keys, SlicePredicate predicate, ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        qosClient.checkLimit();
-
-        Map<ByteBuffer, List<ColumnOrSuperColumn>> result = client.multiget_slice(kvsMethodName, tableRef, keys,
-                predicate, consistency_level);
-        recordBytesRead(() -> getApproximateReadByteCount(result));
-        return result;
+        return qosClient.executeRead(
+                () -> DEFAULT_ESTIMATED_READ_BYTES,
```

Comment:
Should we feed in Cassandra metrics here to estimate the byte read in advance? Otherwise, should we just adjust the ratelimiter after the cassandra call instead of trying to estimate here?

Reply:
we do adjust the rate limiter after the call is done. however, we won't get rate limited unless we ask the ratelimiter for some bandwidth before the request starts